File: ds-algo/midterm/movie_time.py
from collections import deque

class MovieTime:

    movies = deque()
    snacks = deque()

    movieCount = 0
    snackCount = 0

    while movieCount < 3:
        chosenMovies = input("Enter movie "+  str(movieCount + 1) +" of 3: ")
        
        if not chosenMovies:
            print("Please enter a movie title.")
        else:
            movies.append(chosenMovies)
            movieCount += 1
    
    while snackCount < 3:
        chosenSnacks = input("Enter snack " + str(snackCount + 1) + " of 3: ")

        if not chosenSnacks:
            print("Please enter a snack.")
        else:
            snacks.append(chosenSnacks)
            snackCount += 1
    
    print("Movies to watch are: " + str(movies))
    print("Snacks available are: " + str(snacks))

    while snacks:
         action = input("Press S each time you finish a snack.\n")

         if action.upper() == "S":
            snacks.popleft()
            
            if len(snacks) == 0:
                print("No more snacks")
            else:
                print(str(snacks))




   

    

File: ds-algo/midterm/MovieTime.java
import java.util.*;

public class MovieTime {
    
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        String chosenMovies, chosenSnacks;

        int movieCount = 0, snackCount = 0;
        
        Queue<String> movies = new LinkedList<String>();
        Queue<String> snacks = new LinkedList<String>();

        while(movieCount < 3) {
            System.out.print("Enter movie " + (movieCount + 1) + " of 3: ");
            chosenMovies = input.nextLine();
            if(chosenMovies.isEmpty()) {
                System.out.println("Please input a movie title.");
            }else {
                movies.offer(chosenMovies);
                movieCount++;
            }
            
        }

        while(snackCount < 3) {
            System.out.print("Enter snack " + (snackCount + 1) + " of 3: ");
            chosenSnacks = input.nextLine();
            if(chosenSnacks.isEmpty()) {
                System.out.println("Please input a snack.");
            }else {
                snacks.offer(chosenSnacks);
                snackCount++;
            }
            
        }
       

        System.out.println("Movies to watch are: " + movies);
        System.out.println("Snacks available are: " + snacks);

        while(!snacks.isEmpty()) {
            System.out.println("Press S to finish a snack.");
            String action = input.nextLine();

            if(action.equalsIgnoreCase("S")) {
                snacks.poll();

                if(snacks.size() != 0) {
                    System.out.println(snacks);
                }else {
                    System.out.println("No more snacks.");
                }
            }
        }

    }

}
